<?xml version="1.0"?>
<robot name="end_effector" xmlns:xacro="http://www.ros.org/wiki/xacro">

    <link name="davis"/>
    <link name="pressure_ft"/>
    <link name="scancontrol"/>
    <link name="scancontrol_intermediate"/>
    <link name="TCP"/>

    <joint name="TCP_to_cam" type="fixed">
        <parent link="TCP"/>
        <child link="davis"/>
	    <origin xyz="-0.05275792 0.02838982 0.15275118" rpy="-0.0044183 0.0010221 -1.5668429"/>
    </joint>

    <joint name="TCP_to_laser" type="fixed">
        <parent link="TCP"/>
        <child link="scancontrol_intermediate"/>
	    <origin xyz="0.0 0.1 0.15275118" rpy="-0.0044183 0.0010221 -1.5668429"/>
    </joint>

    <joint name="exact_laser_transform" type="fixed">
        <parent link="scancontrol_intermediate"/>
        <child link="scancontrol"/>
	    <origin xyz="0.0 0.0 0.5" rpy="0.0 3.14 0.0"/>
    </joint>

    <joint name="TCP_to_pressure_ft" type="fixed">
        <parent link="TCP"/>
        <child link="pressure_ft"/>
	    <origin xyz="0.05 0 0.15" rpy="0 0 -1.5707963"/>
    </joint>

</robot>


<!-- <origin xyz="-0.05 -0.025 0.08" rpy="0 0 -1.5707963"/> -->